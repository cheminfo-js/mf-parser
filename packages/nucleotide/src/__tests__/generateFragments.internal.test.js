'use strict';

const { generateFragments } = require('..');

describe('test generate internal fragments', () => {
  it('nucleotide HODampDtmpDcmpDgmpDampH internal abw fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      abw: true,
    });
    expect(fragments).toStrictEqual([
      'HODtmpC5H6O4P$w4:a-B3',
      'HODtmpDcmpC5H6O4P$w4:a-B4',
      'HODcmpC5H6O4P$w3:a-B4',
    ]);
  });

  it('nucleotide HODampDtmpDcmpH internal aby fragments', () => {
    let fragments = generateFragments('HODthyDcmpDcmpDgmpDtmpDcmpH', {
      aby: true,
    });
    expect(fragments).toStrictEqual([
      'O-2P-1DcmpC5H6O4P$y5:a-B3',
      'O-2P-1DcmpDcmpC5H6O4P$y5:a-B4',
      'O-2P-1DcmpDcmpDgmpC5H6O4P$y5:a-B5',
      'O-2P-1DcmpC5H6O4P$y4:a-B4',
      'O-2P-1DcmpDgmpC5H6O4P$y4:a-B5',
      'O-2P-1DgmpC5H6O4P$y3:a-B5',
    ]);
  });

  it('nucleotide HODthyDampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpDampDcmpH internal abw fragments', () => {
    let fragments = generateFragments(
      'HODthyDampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpDampDcmpH',
      {
        abw: true,
      },
    );
    expect(fragments).toMatchSnapshot();
  });

  it('nucleotide HODampDtmpDcmpH internal aw fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      aw: true,
    });
    expect(fragments).toStrictEqual([
      'HODtmpDcmpO-1H-1$w4:a3',
      'HODtmpDcmpDgmpO-1H-1$w4:a4',
      'HODcmpDgmpO-1H-1$w3:a4',
    ]);
  });

  it('nucleotide HODampDtmpDcmpH internal bw fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      bw: true,
    });
    expect(fragments).toStrictEqual([
      'HODtmpDcmpH$w4:b3',
      'HODtmpDcmpDgmpH$w4:b4',
      'HODcmpDgmpH$w3:b4',
    ]);
  });

  it('nucleotide to sequence internal modification of HODamp(NH2)DtmpDcmpH with internal fragments', () => {
    let fragments = generateFragments('HODampDampDtmp(NH2)DampH', {
      abw: true,
    });
    expect(fragments).toStrictEqual(['HODampC5H6O4P$w3:a-B3']);
  });
});
