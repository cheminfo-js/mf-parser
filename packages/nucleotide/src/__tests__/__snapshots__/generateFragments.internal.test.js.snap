// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generate internal fragments nucleotide HODthyDampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpDampDcmpH internal abw fragments 1`] = `
Array [
  "HODampC5H6O4P$w12:a3-B",
  "HODampDcmpC5H6O4P$w12:a4-B",
  "HODampDcmpDgmpC5H6O4P$w12:a5-B",
  "HODampDcmpDgmpDtmpC5H6O4P$w12:a6-B",
  "HODampDcmpDgmpDtmpDgmpC5H6O4P$w12:a7-B",
  "HODampDcmpDgmpDtmpDgmpDcmpC5H6O4P$w12:a8-B",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpC5H6O4P$w12:a9-B",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpDampC5H6O4P$w12:a1-B0",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpDampDampC5H6O4P$w12:a1-B1",
  "HODampDcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpC5H6O4P$w12:a1-B2",
  "HODcmpC5H6O4P$w11:a4-B",
  "HODcmpDgmpC5H6O4P$w11:a5-B",
  "HODcmpDgmpDtmpC5H6O4P$w11:a6-B",
  "HODcmpDgmpDtmpDgmpC5H6O4P$w11:a7-B",
  "HODcmpDgmpDtmpDgmpDcmpC5H6O4P$w11:a8-B",
  "HODcmpDgmpDtmpDgmpDcmpDcmpC5H6O4P$w11:a9-B",
  "HODcmpDgmpDtmpDgmpDcmpDcmpDampC5H6O4P$w11:a1-B0",
  "HODcmpDgmpDtmpDgmpDcmpDcmpDampDampC5H6O4P$w11:a1-B1",
  "HODcmpDgmpDtmpDgmpDcmpDcmpDampDampDtmpC5H6O4P$w11:a1-B2",
  "HODgmpC5H6O4P$w10:a5-B",
  "HODgmpDtmpC5H6O4P$w10:a6-B",
  "HODgmpDtmpDgmpC5H6O4P$w10:a7-B",
  "HODgmpDtmpDgmpDcmpC5H6O4P$w10:a8-B",
  "HODgmpDtmpDgmpDcmpDcmpC5H6O4P$w10:a9-B",
  "HODgmpDtmpDgmpDcmpDcmpDampC5H6O4P$w10:a1-B0",
  "HODgmpDtmpDgmpDcmpDcmpDampDampC5H6O4P$w10:a1-B1",
  "HODgmpDtmpDgmpDcmpDcmpDampDampDtmpC5H6O4P$w10:a1-B2",
  "HODtmpC5H6O4P$w9:a6-B",
  "HODtmpDgmpC5H6O4P$w9:a7-B",
  "HODtmpDgmpDcmpC5H6O4P$w9:a8-B",
  "HODtmpDgmpDcmpDcmpC5H6O4P$w9:a9-B",
  "HODtmpDgmpDcmpDcmpDampC5H6O4P$w9:a1-B0",
  "HODtmpDgmpDcmpDcmpDampDampC5H6O4P$w9:a1-B1",
  "HODtmpDgmpDcmpDcmpDampDampDtmpC5H6O4P$w9:a1-B2",
  "HODgmpC5H6O4P$w8:a7-B",
  "HODgmpDcmpC5H6O4P$w8:a8-B",
  "HODgmpDcmpDcmpC5H6O4P$w8:a9-B",
  "HODgmpDcmpDcmpDampC5H6O4P$w8:a1-B0",
  "HODgmpDcmpDcmpDampDampC5H6O4P$w8:a1-B1",
  "HODgmpDcmpDcmpDampDampDtmpC5H6O4P$w8:a1-B2",
  "HODcmpC5H6O4P$w7:a8-B",
  "HODcmpDcmpC5H6O4P$w7:a9-B",
  "HODcmpDcmpDampC5H6O4P$w7:a1-B0",
  "HODcmpDcmpDampDampC5H6O4P$w7:a1-B1",
  "HODcmpDcmpDampDampDtmpC5H6O4P$w7:a1-B2",
  "HODcmpC5H6O4P$w6:a9-B",
  "HODcmpDampC5H6O4P$w6:a1-B0",
  "HODcmpDampDampC5H6O4P$w6:a1-B1",
  "HODcmpDampDampDtmpC5H6O4P$w6:a1-B2",
  "HODampC5H6O4P$w5:a1-B0",
  "HODampDampC5H6O4P$w5:a1-B1",
  "HODampDampDtmpC5H6O4P$w5:a1-B2",
  "HODampC5H6O4P$w4:a1-B1",
  "HODampDtmpC5H6O4P$w4:a1-B2",
  "HODtmpC5H6O4P$w3:a1-B2",
]
`;
